File: snippets/ign/ign.js
// Axios

import axios from "axios";

const options = {
  method: 'PUT',
  url: 'https://pd.{Shard}.a.pvp.net/name-service/v2/players',
  headers: {'Content-Type': 'application/json'},
  data: ['PUUID', 'OTHERPUUID']
};

axios.request(options).then(function (response) {
  console.log(response.data);
}).catch(function (error) {
  console.error(error);
});

//Fetch

fetch("https://pd.{Shard}.a.pvp.net/name-service/v2/players", {
  "method": "PUT",
  "headers": {
    "Content-Type": "application/json"
  },
  "body": "[\"PUUID\",\"OTHERPUUID\"]"
})
.then(response => {
  console.log(response);
})
.catch(err => {
  console.error(err);
});

File: snippets/ign/ign.py
#Requests

import requests

url = "https://pd.{Shard}.a.pvp.net/name-service/v2/players"

payload = ["{PUUID}", "{OtherPUUID}}"]
headers = {"Content-Type": "application/json"}

response = requests.request("PUT", url, json=payload, headers=headers)

print(response.text)

#HTTP client
import http.client

conn = http.client.HTTPSConnection("pd.{Shard}.a.pvp.net")

payload = "[\"PUUID\", \"OTHERPUUID\"]"

headers = { 'Content-Type': "application/json" }

conn.request("PUT", "/name-service/v2/players", payload, headers)

res = conn.getresponse()
data = res.read()

print(data.decode("utf-8"))
